lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -274,8 +274,8 @@ function _campaign_resource_signup($nid, $values) {
     unset($values['northstar_id']);
   }
 
-  // If we pass either a Northstar ID or Drupal UID to `uid`, convert it to a real UID.
-  $values['uid'] = dosomething_user_convert_to_legacy_id($values['uid']);
+  // If a Drupal UID is passed to `uid`, convert it to the user's Northstar ID.
+  $values['uid'] = dosomething_user_convert_to_northstar_id($values['uid']);
```

Comment:
Oh no, that definitely makes things more complicated. 😩 We may want to hold off on this work until we've started sending these notifications via Blink/Customer.io.

> Is the idea here to just make sure the user's signup and reportback gets logged in Rogue?

We want to remove the requirement that every user _has to have_ a Phoenix UID, since that allows users who don't have Phoenix UIDs to signup or reportback for campaigns via Gambit. (Normally, Northstar makes a HTTP request to Phoenix to create the Drupal UID whenever a user is created but that sometimes breaks... I think due to network issues.)

Ideally, we should only have to make a Drupal UID when a user first logs in to Phoenix via the web, and no other systems should rely on it. This makes Northstar/Rogue more resilient, and makes it easier for us to remove Phoenix altogether one day.